Failed build with GCC 7

Which is default compiler in upcoming Fedora 26

```
libtool: compile:  g++ -Wno-narrowing -std=c++11 -I. -I/builddir/build/BUILD/php-pecl-v8js-1.3.4/NTS -DPHP_ATOM_INC -I/builddir/build/BUILD/php-pecl-v8js-1.3.4/NTS/include -I/builddir/build/BUILD/php-pecl-v8js-1.3.4/NTS/main -I/builddir/build/BUILD/php-pecl-v8js-1.3.4/NTS -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib -I/usr -DHAVE_CONFIG_H -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic -c /builddir/build/BUILD/php-pecl-v8js-1.3.4/NTS/v8js_commonjs.cc  -fPIC -DPIC -o .libs/v8js_commonjs.o
In file included from /usr/include/dirent.h:245:0,
                 from /usr/include/php/Zend/zend_virtual_cwd.h:91,
                 from /usr/include/php/main/php.h:411,
                 from /builddir/build/BUILD/php-pecl-v8js-1.3.4/NTS/php_v8js_macros.h:39,
                 from /builddir/build/BUILD/php-pecl-v8js-1.3.4/NTS/v8js.cc:18:
/builddir/build/BUILD/php-pecl-v8js-1.3.4/NTS/v8js_class.h: In function 'v8js_ctx* ```
v8js_ctx_fetch_object(zend_object*)':
/builddir/build/BUILD/php-pecl-v8js-1.3.4/NTS/v8js_class.h:92:54: warning: offsetof within non-standard-layout type 'v8js_ctx' is undefined [-Winvalid-offsetof]
  return (struct v8js_ctx *)((char *)obj - XtOffsetOf(struct v8js_ctx, std));
/builddir/build/BUILD/php-pecl-v8js-1.3.4/NTS/v8js_class.h:92:43: note: in expansion of macro 'XtOffsetOf'
  return (struct v8js_ctx *)((char *)obj - XtOffsetOf(struct v8js_ctx, std));
                                           ^~~~~~~~~~
In file included from /builddir/build/BUILD/php-pecl-v8js-1.3.4/NTS/php_v8js_macros.h:52:0,
                 from /builddir/build/BUILD/php-pecl-v8js-1.3.4/NTS/v8js.cc:18:
/builddir/build/BUILD/php-pecl-v8js-1.3.4/NTS/v8js_v8.h: At global scope:
/builddir/build/BUILD/php-pecl-v8js-1.3.4/NTS/v8js_v8.h:51:12: error: 'std::function' has not been declared
       std::function< v8::Local<v8::Value>(v8::Isolate *) >& v8_call TSRMLS_DC);
            ^~~~~~~~
/builddir/build/BUILD/php-pecl-v8js-1.3.4/NTS/v8js_v8.h:51:20: error: expected ',' or '...' before '<' token
       std::function< v8::Local<v8::Value>(v8::Isolate *) >& v8_call TSRMLS_DC);
                    ^
make: *** [Makefile:205: v8js.lo] Error 1
make: *** Waiting for unfinished jobs....

```

--- a/Vagrantfile
+++ b/Vagrantfile
@@ -10,32 +10,55 @@ Vagrant.configure("2") do |config|
   # boxes at https://atlas.hashicorp.com/search.
   config.vm.box = "ubuntu/trusty64"
 
-  # Share an additional folder to the guest VM. The first argument is
-  # the path on the host to the actual folder. The second argument is
-  # the path on the guest to mount the folder. And the optional third
-  # argument is a set of non-required options.
-  config.vm.synced_folder ".", "/data/v8js"
-
   config.vm.provider "lxc" do |lxc, override|
     override.vm.box = "fgrehm/trusty64-lxc"
   end
 
-  config.vm.provision "shell", inline: <<-SHELL
-    gpg --keyserver keys.gnupg.net --recv 7F438280EF8D349F
-    gpg --armor --export 7F438280EF8D349F | apt-key add -
 
-    apt-get update
-    apt-get install -y software-properties-common gdb tmux git tig curl apache2-utils
-    add-apt-repository ppa:ondrej/php
-  SHELL
-
+  #
+  # mass-define "generic" Ubuntu boxes
+  #
   %w{5.1 5.2 5.4 5.7 5.8 5.9}.each { |version|
     config.vm.define "v8-#{version}" do |i|
+      i.vm.synced_folder ".", "/data/v8js"
+
       i.vm.provision "shell", inline: <<-SHELL
+      gpg --keyserver keys.gnupg.net --recv 7F438280EF8D349F
+      gpg --armor --export 7F438280EF8D349F | apt-key add -
+
+      apt-get update
+      apt-get install -y software-properties-common gdb tmux git tig curl apache2-utils
+
+      add-apt-repository ppa:ondrej/php
       add-apt-repository ppa:pinepain/libv8-#{version}
       apt-get update
       apt-get install -y php7.1-dev libv8-#{version}-dbg libv8-#{version}-dev
     SHELL
     end
   }
+
+
+  #
+  # Fedora-based box with GCC7, V8 5.2 + PHP 7.1 installed
+  # (primarily to reproduce #294)
+  #
+  config.vm.define "fedora-26-gcc7" do |i|
+    i.vm.box = "vbenes/fedora-rawhide-server"
+    i.ssh.insert_key = false
+
+    # unfortunately vboxsf isn't currently available (due to an issue with the base image)
+    # therefore fall back to nfs
+    i.vm.synced_folder ".", "/data/v8js", type: "nfs"
+    i.vm.network "private_network", ip: "192.168.50.2"
+
+    i.vm.provision "shell", inline: <<-SHELL
+      dnf -y update
+      dnf -y install gcc-c++ gdb tmux git tig curl vim
+      dnf -y install v8-devel php-devel
+    SHELL
+  end
+
+  config.vm.provision "shell", inline: <<-SHELL
+    mkdir -p /data/build && chown vagrant. /data/build
+  SHELL
 end

--- a/v8js_v8.h
+++ b/v8js_v8.h
@@ -14,6 +14,8 @@
 #ifndef V8JS_V8_H
 #define V8JS_V8_H
 
+#include <functional>
+
 /* Helper macros */
 #define V8JS_SYM(v)			v8::String::NewFromUtf8(isolate, v, v8::String::kInternalizedString, sizeof(v) - 1)
 #define V8JS_SYML(v, l)		v8::String::NewFromUtf8(isolate, v, v8::String::kInternalizedString, l)

